src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/DoNotIgnoreMethodResults.cs

```diff
@@ -110,68 +110,193 @@ public override void Initialize(AnalysisContext analysisContext)
+                                }
+                                break;
+
+                            case OperationKind.Invocation:
+                                IInvocationOperation invocationExpression = ((IInvocationOperation)expression);
+                                IMethodSymbol targetMethod = invocationExpression.TargetMethod;
+                                if (targetMethod == null)
+                                {
+                                    break;
+                                }
+
+                                if (IsStringCreatingMethod(targetMethod))
+                                {
+                                    rule = StringCreationRule;
+                                }
+                                else if (IsTryParseMethod(targetMethod))
+                                {
+                                    rule = TryParseRule;
+                                }
+                                else if (IsHResultOrErrorCodeReturningMethod(targetMethod))
+                                {
+                                    rule = HResultOrErrorCodeRule;
+                                }
+                                else if (IsPureMethod(targetMethod, opContext.Compilation))
+                                {
+                                    rule = PureMethodRule;
+                                }
+
+                                targetMethodName = targetMethod.Name;
+                                break;
+                        }
+
+                        if (rule != null)
+                        {
+                            Diagnostic diagnostic = Diagnostic.Create(rule, expression.Syntax.GetLocation(), method.Name, targetMethodName);
+                            opContext.ReportDiagnostic(diagnostic);
+                        }
+                    }, OperationKind.ExpressionStatement);
+                });
             });
         }
 
+        private static bool ShouldSkipAnalyzing(OperationAnalysisContext operationContext, INamedTypeSymbol expectedExceptionType)
+        {
+            bool IsThrowsArgument(IParameterSymbol parameterSymbol, string argumentName, ImmutableHashSet<string> methodNames, string containingSymbol)
+            {
+                return parameterSymbol.Name == argumentName &&
+                       parameterSymbol.ContainingSymbol is IMethodSymbol methodSymbol &&
+                       methodNames.Contains(methodSymbol.Name) &&
+                       methodSymbol.ContainingSymbol.ToDisplayString() == containingSymbol;
+            }
+
+            bool IsNUnitThrowsArgument(IParameterSymbol parameterSymbol)
+            {
+                var methodNames = ImmutableHashSet.Create(new[]
+                {
+                    "Throws",
+                    "Catch",
+                    "DoesNotThrow",
+                    "ThrowsAsync",
+                    "CatchAsync",
+                    "DoesNotThrowAsync"
+                });
```

Comment:
Should not we move this HashSet to private static readonly? And the same for the one below.

Reply:
+1

---
In reply to: [162737270](https://github.com/dotnet/roslyn-analyzers/pull/1539#discussion_r162737270) [](ancestors = 162737270)